cpp/src/io/json/nested_json_gpu.cu

```diff
@@ -330,47 +330,115 @@ enum class dfa_symbol_group_id : uint8_t {
+      translation_table{
+        {// IN_STATE       {      [      }      ]    "   \ <delim> OTHER
+         // TT_OOS     {{{'{'}, {'['}, {'}'}, {']'}, {}, {}, {}, {}}},
+         // TT_STR     {{{}, {}, {}, {}, {}, {}, {}, {}}},
+         // TT_ESC     {{{}, {}, {}, {}, {}, {}, {}, {}}}}};
+
+      // Translation table for the JSON lines format that recovers from invalid JSON lines
+      std::array<std::array<std::vector<SymbolT>, NUM_SYMBOL_GROUPS>, TT_NUM_STATES>
+        resetting_translation_table{
+          {// IN_STATE       {      [      }      ]      "      \     \n    OTHER
+           // TT_OOS    {{{'{'}, {'['}, {'}'}, {']'}, {}, {}, {'\n'}, {}}},
+           // TT_STR    {{{}, {}, {}, {}, {}, {}, {'\n'}, {}}},
+           // TT_ESC    {{{}, {}, {}, {}, {}, {}, {'\n'}, {}}}}};
+    */
+    if (state_id == static_cast<StateT>(dfa_states::TT_STR) ||
+        state_id == static_cast<StateT>(dfa_states::TT_ESC)) {
+      if (stack_behavior == stack_behavior_t::ResetOnDelimiter &&
+          match_id == static_cast<SymbolGroupT>(dfa_symbol_group_id::DELIMITER_CHAR))
+        return read_symbol;
+    }
+    if (match_id == static_cast<SymbolGroupT>(dfa_symbol_group_id::OPENING_BRACE) ||
+        match_id == static_cast<SymbolGroupT>(dfa_symbol_group_id::OPENING_BRACKET) ||
+        match_id == static_cast<SymbolGroupT>(dfa_symbol_group_id::CLOSING_BRACE) ||
+        match_id == static_cast<SymbolGroupT>(dfa_symbol_group_id::CLOSING_BRACKET))
+      return read_symbol;
+    if (match_id == static_cast<SymbolGroupT>(dfa_symbol_group_id::DELIMITER_CHAR) &&
+        stack_behavior == stack_behavior_t::ResetOnDelimiter)
+      return read_symbol;
+
+    return 0;
+  }
+
+  /**
+   * @brief Returns the number of output characters for a given transition.
+   */
+  template <typename StateT, typename SymbolGroupT, typename SymbolT>
+  constexpr CUDF_HOST_DEVICE uint32_t operator()(StateT const state_id,
+                                                 SymbolGroupT const match_id,
+                                                 SymbolT const read_symbol) const
+  {
+    if (state_id == static_cast<StateT>(dfa_states::TT_STR) ||
+        state_id == static_cast<StateT>(dfa_states::TT_ESC)) {
+      if (stack_behavior == stack_behavior_t::ResetOnDelimiter &&
+          match_id == static_cast<SymbolGroupT>(dfa_symbol_group_id::DELIMITER_CHAR))
+        return 1;
+    }
+    if (match_id == static_cast<SymbolGroupT>(dfa_symbol_group_id::OPENING_BRACE) ||
+        match_id == static_cast<SymbolGroupT>(dfa_symbol_group_id::OPENING_BRACKET) ||
+        match_id == static_cast<SymbolGroupT>(dfa_symbol_group_id::CLOSING_BRACE) ||
+        match_id == static_cast<SymbolGroupT>(dfa_symbol_group_id::CLOSING_BRACKET))
+      return 1;
+    if (match_id == static_cast<SymbolGroupT>(dfa_symbol_group_id::DELIMITER_CHAR) &&
+        stack_behavior == stack_behavior_t::ResetOnDelimiter)
+      return 1;
+
+    return 0;
+  }
+};
```

Comment:
I haven't compared to the original translation table, but couldn't we just keep the original translation table?

Reply:
That would save you some (minor) work on modifications on the logical stack. 

If it should turn out that we want to translate `delim`to `delim`, I would suggest to rather modify the corresponding entries of `translation_table` instead of implementing it all as a function object.

Reply:
Thank you for the suggestions! 
Similar to the `get_sgid_lut` for the symbol groups lookup table, I have added `get_transition_table` and `get_translation_table` functions that constructs and returns the correct table depending on the stack behaviour option and the delimiter.